cartographer_grpc/framework/rpc.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2017 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CARTOGRAPHER_GRPC_FRAMEWORK_RPC_H
+#define CARTOGRAPHER_GRPC_FRAMEWORK_RPC_H
+
+#include <memory>
+#include <mutex>
+#include <unordered_set>
+
+#include "cartographer_grpc/framework/rpc_handler.h"
+#include "grpc++/grpc++.h"
+
+namespace cartographer_grpc {
+namespace framework {
+
+class Service;
+class Rpc {
+ public:
+  enum class State { NEW_CONNECTION = 0, READ, WRITE, DONE };
+  struct RpcState {
+    const State state;
+    Rpc* rpc;
+    void* GetTag() { return reinterpret_cast<void*>(this); }
```

Comment:
Is this really needed? Can't you just pass RpcState* to a function that accepts void*?

Reply:
Removed.